### PR TITLE
test_setup: Enhance code of create private bridge

### PR DESCRIPTION
**Pull request by PaulYuuu**
1. Check if dnsmasq package exists before _start_dhcp_server
2. Use a better method to check ovs bridge
    - If the bridge name contains digits, the bridge info is similar to the following:
         ```bash
        $ ovs-vsctl show
        65fbb3f9-b98f-4009-8c20-8eb725ba3d2a
            Bridge ovsI
                Port ovsI
                    Interface ovsI
                        type: internal
            Bridge "ovs0"
                Port "ovs0"
                    Interface "ovs0"
                        type: internal
            ovs_version: "2.11.0"
        ```
    - So `re.match` could not match it
https://github.com/avocado-framework/avocado-vt/blob/9acbffa1bd200be205c5cff771c0a1b3e277a80a/virttest/test_setup.py#L1005-L1012
    - This will block the call to the cleanup method
https://github.com/avocado-framework/avocado-vt/blob/9acbffa1bd200be205c5cff771c0a1b3e277a80a/virttest/test_setup.py#L981-L984

ID: 1783859
Signed-off-by: Yihuang Yu <yihyu@redhat.com>

### CODE REVIEW

File: virttest/test_setup.py [hunk 1]
```diff
@@ -868,6 +868,8 @@ def _enable_nat(self):
             self._iptables_add(rule)
 
     def _start_dhcp_server(self):
+        if not utils_package.package_install("dnsmasq"):
+            exceptions.TestSkipError("Failed to install dnsmasq package")
         process.system("service dnsmasq stop")
```
Comment (xutian) [inline virttest/test_setup.py]: let's stop dnsmasq service with service module in framework  to make code works more stable. eg, 
`
 service.SpecificServiceManager("dnsmasq").stop()`

Other parts looks good to me.

Comment (PaulYuuu) [reply virttest/test_setup.py]: > let's stop dnsmasq service with service module in framework to make code works more stable. eg,
> ` service.SpecificServiceManager("dnsmasq").stop()`
> 
Updated.
> Other parts looks good to me.

Also, I replaced some steps with the Bridge Manager method, test pass with `netdst=private` and `testcase=boot`. Please help to review again, thanks.

**Comment by xutian** [comment]
Acked-by: XuTian <xutian@redhat.com>

**Comment by xutian** [comment]
@PaulYuuu   you need to read trigger CI test.

**Comment by PaulYuuu** [comment]
> @PaulYuuu you need to read trigger CI test.

Fixed, thanks for your reminder.

**Comment by xutian** [comment]
Acked-by: XuTian <xutian@redhat.com>

**Comment by PaulYuuu** [comment]
@vivianQizhu Could you help to review? Thanks a lot.

File: virttest/test_setup.py [hunk 2]
```diff
@@ -868,7 +872,9 @@ def _enable_nat(self):
             self._iptables_add(rule)
 
     def _start_dhcp_server(self):
-        process.system("service dnsmasq stop")
+        if not utils_package.package_install("dnsmasq"):
```
Comment (xutian) [inline virttest/test_setup.py]: only need to install package if it's not installed.

File: virttest/test_setup.py [hunk 3]
```diff
@@ -975,8 +972,10 @@ def _disable_nat(self):
             self._iptables_del(rule)
 
     def _remove_bridge(self):
-        process.system("ip link delete %s type bridge" % self.brname,
-                       ignore_status=True)
+        try:
+            self.bridge_manager.del_bridge(self.brname)
+        except:
```
Comment (xutian) [inline virttest/test_setup.py]: better to print warning here

Comment (PaulYuuu) [reply virttest/test_setup.py]: updated

File: virttest/test_setup.py (hunk 2 above)
Comment (PaulYuuu) [reply virttest/test_setup.py]: `package_install` will check it, so don't need to double check.

**Comment by PaulYuuu** [comment]
@yanglei-rh Please help to review, thanks.

**Pull request merged**